sdk/communication/AzureCommunicationChat/Source/ChatThreadClient.swift

```diff
@@ -0,0 +1,348 @@
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import AzureCommunication
+import AzureCore
+import Foundation
+
+public class ChatThreadClient {
+    // MARK: Properties
+
+    public let threadId: String
+    private let endpoint: String
+    private let credential: CommunicationUserCredential
+    private let options: AzureCommunicationChatClientOptions
+    private let service: AzureCommunicationChatService
+
+    // MARK: Initializers
+
+    /// Create a ChatThreadClient.
+    /// - Parameters:
+    ///   - endpoint: The Communication Services endpoint.
+    ///   - credential: The user credential.
+    ///   - threadId: The chat thread id.
+    ///   - options: Options used to configure the client.
+    public init(
+        endpoint: String,
+        credential: CommunicationUserCredential,
+        threadId: String,
+        withOptions options: AzureCommunicationChatClientOptions
+    ) throws {
+        self.threadId = threadId
+        self.endpoint = endpoint
+        self.credential = credential
+        self.options = options
+
+        guard let endpointUrl = URL(string: endpoint) else {
+            throw AzureError.client("Unable to form base URL")
+        }
+
+        let authPolicy = CommunicationUserCredentialPolicy(credential: credential)
+
+        let client = try AzureCommunicationChatClient(
+            endpoint: endpointUrl,
+            authPolicy: authPolicy,
+            withOptions: options
+        )
+
+        self.service = client.azureCommunicationChatService
+    }
+
+    // MARK: Public Methods
+
+    /// Updates the ChatThread's properties.
+    /// - Parameters:
+    ///   - chatThread: Request for updating a chat thread.
+    ///   - options: Update chat thread options.
+    ///   - completionHandler: A completion handler that receives a status code on success.
+    public func update(
+        chatThread: UpdateChatThreadRequest,
+        withOptions options: AzureCommunicationChatService.UpdateChatThreadOptions? = nil,
+        completionHandler: @escaping HTTPResultHandler<Void>
+    ) {
+        service.update(chatThread: chatThread, chatThreadId: threadId, withOptions: options) { result, httpResponse in
+            switch result {
+            case .success:
+                completionHandler(.success(()), httpResponse)
+
+            case let .failure(error):
+                completionHandler(.failure(error), httpResponse)
+            }
+        }
+    }
+
+    /// Sends a read receipt.
+    /// - Parameters:
+    ///   - request: Request for sending a read receipt.
+    ///   - options: Send read receipt options.
+    ///   - completionHandler: A completion handler that receives a status code on success.
+    public func send(
+        readReceipt: SendReadReceiptRequest,
```

Comment:
Consider using just `ReadReceipt` here.

Reply:
Commented on Chatmessage and sendreadreceipt, will leave as is